src/process.cr

```diff
@@ -428,31 +446,33 @@ class Process
 
     Dir.cd(chdir) if chdir
 
-    LibC.execvp(command, argv) != -1
+    argv = [command.check_no_null_byte.to_unsafe]
+    args.try &.each do |arg|
+      argv << arg.check_no_null_byte.to_unsafe
+    end
+    argv << Pointer(UInt8).null
+
+    LibC.execvp(command, argv)
+    raise Errno.new("execvp")
```

Comment:
Could we avoid the overhead of raising an exception that is immediately caught? That's how it was before.
Could still have an handler for other `Errno` raised in `reopen_io`.

Reply:
I don't think it's worth the effort. An exception will still be raised either way, and it just duplicates code to have to check for errno in two ways.